src/rust/engine/rule_graph/src/tests.rs

```diff
@@ -304,6 +304,31 @@ fn in_scope_computed() {
   graph.find_root_edges(vec![], "a").unwrap();
 }
 
+#[test]
+fn in_scope_provided() {
```

Comment:
Should this be cherry-picked?

Reply:
`in_scope_types` is only on `main` so far.